File: loan-backend/loan-backend/src/main/java/com/loan/backend/service/LeadService.java
package com.loan.backend.service;

import com.loan.backend.dto.CreateLeadRequest;
import com.loan.backend.dto.UpdateLeadStatusRequest;
import com.loan.backend.entity.Lead;

import java.util.List;

public interface LeadService {

    Lead createLead(CreateLeadRequest request);
    List<Lead> getAllLeads();
    Lead getLeadById(Long id);
    Lead updateLeadStatus(Long id, UpdateLeadStatusRequest request);
    void deleteLead(Long id);
}

File: loan-frontend/src/services/leadService.js
import api from "../api";

export const getAllLeads = async () => {
  const response = await api.get("/api/leads");
  return response.data;
};

export const getLeadById = async (id) => {
  const response = await api.get(`/api/leads/${id}`);
  return response.data;
};

export const updateLeadStatus = async (id, status) => {
  const response = await api.put(`/api/leads/${id}/status`, {status});
  return response.data;
};

export const createLead = async (lead) => {
  const response = await api.post("/api/leads", lead);
  return response.data;
};

export const deleteLead = async (id) => {
  const response = await api.delete(`/api/leads/${id}`);
  return response.data;
};